go/actions/test/action.yml

```diff
@@ -0,0 +1,76 @@
+name: Test go extractor
+description: Run build, QL tests and optionally basic code sanity checks (formatting and generation)
+inputs:
+  go-version:
+    description: Which Go version to use for running the tests
+    required: false
+    default: ~1.22.0
+  run-code-checks:
+    description: Whether to run formatting, code and qhelp generation checks
+    required: false
+    default: false
+runs:
+  using: composite
+  steps:
+    - name: Set up Go
+      uses: actions/setup-go@v5
+      with:
+        go-version: ${{ inputs.go-version }}
+        cache: false
+      id: go
+
+    - name: Set up CodeQL CLI
+      uses: ./.github/actions/fetch-codeql
+
+    - name: Enable problem matchers in repository
+      shell: bash
+      run: 'find .github/problem-matchers -name \*.json -exec echo "::add-matcher::{}" \;'
+
+    - name: Build
+      shell: bash
+      run: |
+        bazel run go:create-extractor-pack
+
+    - name: Check that all Go code is autoformatted
+      if: inputs.run-code-checks == 'true'
+      shell: bash
+      run: |
+          cd go
+          make check-formatting
+
+    - name: Check checked-in generated code
+      if: inputs.run-code-checks == 'true'
+      shell: bash
+      run: |
+        bazel run go:gen
+        git add .
+        git diff --exit-code HEAD || (
+          echo "please run bazel run //go:gen"
+          exit 1
+        )
```

Comment:
Would it make sense to change the conditions to `inputs.run-code-checks == 'true' && always()` so that they run, even if a previous step failed? That way the checks would be performed, even if the build / a previous check fails and save us from potentially needing multiple CI runs / incremental fixes. (Would need to check that the proposed conditions work as intended, since `always()` is sort of magical.)

Reply:
Good suggestion, but I guess that this shouldn't be required if we move this check up as you suggested

Reply:
I think it would probably still make sense to have for the "Check checked-in generated code" step, since it would be nice to run it even if "Check that all Go code is autoformatted" fails so that potentially two issues can become apparent from one CI run.